This patch disables prinstsupport, sql, concurrent, testlib  base libs compilation
--- a/qtbase/src/src.pro
+++ b/qtbase/src/src.pro
@@ -66,22 +66,10 @@ src_dbus.target = sub-dbus
 src_dbus.depends = src_corelib
 force_dbus_bootstrap: src_dbus.depends += src_tools_bootstrap_dbus  # avoid syncqt race
 
-src_concurrent.subdir = $$PWD/concurrent
-src_concurrent.target = sub-concurrent
-src_concurrent.depends = src_corelib
-
-src_sql.subdir = $$PWD/sql
-src_sql.target = sub-sql
-src_sql.depends = src_corelib
-
 src_network.subdir = $$PWD/network
 src_network.target = sub-network
 src_network.depends = src_corelib
 
-src_testlib.subdir = $$PWD/testlib
-src_testlib.target = sub-testlib
-src_testlib.depends = src_corelib   # testlib links only to corelib, but see below for the headers
-
 src_3rdparty_pcre2.subdir = $$PWD/3rdparty/pcre2
 src_3rdparty_pcre2.target = sub-3rdparty-pcre2
 
@@ -125,13 +113,6 @@ src_openglextensions.subdir = $$PWD/open
 src_openglextensions.target = sub-openglextensions
 src_openglextensions.depends = src_gui
 
-src_printsupport.subdir = $$PWD/printsupport
-src_printsupport.target = sub-printsupport
-src_printsupport.depends = src_corelib src_gui src_widgets src_tools_uic
-
-src_plugins.subdir = $$PWD/plugins
-src_plugins.target = sub-plugins
-
 src_android.subdir = $$PWD/android
 
 # this order is important
@@ -152,14 +133,8 @@ TOOLS = src_tools_moc src_tools_rcc src_
 win32:SUBDIRS += src_winmain
 qtConfig(network) {
     SUBDIRS += src_network
-    src_plugins.depends += src_network
-}
-qtConfig(sql) {
-    SUBDIRS += src_sql
-    src_plugins.depends += src_sql
 }
 qtConfig(xml): SUBDIRS += src_xml
-qtConfig(testlib): SUBDIRS += src_testlib
 qtConfig(dbus) {
     force_dbus_bootstrap|qtConfig(private_tests): \
         SUBDIRS += src_tools_bootstrap_dbus
@@ -167,9 +142,7 @@ qtConfig(dbus) {
     TOOLS += src_tools_qdbusxml2cpp src_tools_qdbuscpp2xml
     qtConfig(accessibility-atspi-bridge): \
         src_platformsupport.depends += src_dbus src_tools_qdbusxml2cpp
-    src_plugins.depends += src_dbus src_tools_qdbusxml2cpp src_tools_qdbuscpp2xml
 }
-qtConfig(concurrent): SUBDIRS += src_concurrent
 qtConfig(gui) {
     qtConfig(harfbuzz):!qtConfig(system-harfbuzz) {
         SUBDIRS += src_3rdparty_harfbuzzng
@@ -190,24 +163,17 @@ qtConfig(gui) {
     }
     SUBDIRS += src_gui src_platformsupport src_platformheaders
     qtConfig(opengl): SUBDIRS += src_openglextensions
-    src_plugins.depends += src_gui src_platformsupport src_platformheaders
-    src_testlib.depends += src_gui      # if QtGui is enabled, QtTest requires QtGui's headers
     qtConfig(widgets) {
         SUBDIRS += src_tools_uic src_widgets
         !android-embedded: SUBDIRS += src_printsupport
         TOOLS += src_tools_uic
-        src_plugins.depends += src_widgets
-        !android-embedded: src_plugins.depends += src_printsupport
-        src_testlib.depends += src_widgets        # if QtWidgets is enabled, QtTest requires QtWidgets's headers
         qtConfig(opengl) {
             SUBDIRS += src_opengl
-            src_plugins.depends += src_opengl
         }
     }
 }
-SUBDIRS += src_plugins
 
-nacl: SUBDIRS -= src_network src_testlib
+nacl: SUBDIRS -= src_network
 
 android:!android-embedded: SUBDIRS += src_android src_3rdparty_gradle
 
